src/ConnectedMode/Migration/VsAwareFileSystem.cs

```diff
@@ -58,52 +61,148 @@ internal class VsAwareFileSystem : IVsAwareFileSystem
             this.threadHandling = threadHandling;
         }
 
-        public Task DeleteFolderAsync(string folderPath)
+        public async Task DeleteFolderAsync(string folderPath)
         {
-            // TODO - error handling
+            Debug.Assert(batchStarted, "Expecting the changes to happen in a batch");   
+
             logger.LogMigrationVerbose($"Deleting directory: {folderPath}");
+
+            await CheckOutFilesToDeleteAsync(folderPath);
+
             fileSystem.Directory.Delete(folderPath, true);
-            return Task.CompletedTask;
         }
 
         public Task<string> LoadAsTextAsync(string filePath)
         {
-            // TODO - error handling
             logger.LogMigrationVerbose($"Reading file: {filePath}");
             var content = fileSystem.File.ReadAllText(filePath);
             return Task.FromResult(content);
         }
 
-        public Task SaveAsync(string filePath, string text)
+        public async Task SaveAsync(string filePath, string text)
         {
-            // TODO - error handling
+            Debug.Assert(batchStarted, "Expecting the changes to happen in a batch");
+
             logger.LogMigrationVerbose($"Saving file: {filePath}");
+
+            bool success = await CheckoutForEditAsync(filePath);
```

Comment:
Before we save a file we need to tell VS we are going to check it out.
If we can't check it out we will throw, which will cause the migration to end.